istioctl/pkg/proxyconfig/proxyconfig.go

```diff
@@ -397,6 +400,9 @@ func allConfigCmd(ctx cli.Context) *cobra.Command {
 		Example: `  # Retrieve summary about all configuration for a given pod from Envoy.
   istioctl proxy-config all <pod-name[.namespace]>
 
+  # Retrieve summary about all configuration for a pod under a deployment
+  istioctl proxy-config all deployment/<deployment-name[.namespace]> from Envoy.
+
```

Comment:
nit: sentence should be ordered similar to the other comments you added:
```suggestion
  # Retrieve summary about all configuration for a pod under a deployment from Envoy.
  istioctl proxy-config all deployment/<deployment-name[.namespace]>
```